src/Ledger/Ratify.lagda

```diff
@@ -35,6 +37,7 @@ record StakeDistrs : Set where
 record RatifyEnv : Set where
   field stakeDistrs   : StakeDistrs
         currentEpoch  : Epoch
+        ccHotKeys     : KeyHash ↛ Maybe KeyHash
```

Comment:
Why does this map to a `Maybe KeyHash`?

Reply:
This is the mechanism for resigning. If you put a `nothing` in there, you have resigned & you can't un-resign anymore.